applications/RomApplication/custom_strategies/lspg_rom_builder_and_solver.h

```diff
@@ -0,0 +1,478 @@
+        // this->AssignSettings(this_parameters_copy);
+    }
+
+    ~LSPGROMBuilderAndSolver() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void SetUpDofSet(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart &rModelPart) override
+    {
+        KRATOS_TRY;
+
+        KRATOS_INFO_IF("LSPGROMBuilderAndSolver", (this->GetEchoLevel() > 1)) << "Setting up the dofs" << std::endl;
+        KRATOS_INFO_IF("LSPGROMBuilderAndSolver", (this->GetEchoLevel() > 2)) << "Number of threads" << ParallelUtilities::GetNumThreads() << "\n" << std::endl;
+        KRATOS_INFO_IF("LSPGROMBuilderAndSolver", (this->GetEchoLevel() > 2)) << "Initializing element loop" << std::endl;
+
+        // Get model part data
+        if (this->mHromWeightsInitialized == false) {
+            this->InitializeHROMWeights(rModelPart);
+        }
+
+        auto dof_queue = this->ExtractDofSet(pScheme, rModelPart);
+
+        // Fill a sorted auxiliary array of with the DOFs set
+        KRATOS_INFO_IF("LSPGROMBuilderAndSolver", (this->GetEchoLevel() > 2)) << "Initializing ordered array filling\n" << std::endl;
+        auto dof_array = this->SortAndRemoveDuplicateDofs(dof_queue);
+
+        // Update base builder and solver DOFs array and set corresponding flag
+        BaseType::GetDofSet().swap(dof_array);
+        BaseType::SetDofSetIsInitializedFlag(true);
+
+        // Throw an exception if there are no DOFs involved in the analysis
+        KRATOS_ERROR_IF(BaseType::GetDofSet().size() == 0) << "No degrees of freedom!" << std::endl;
+        KRATOS_INFO_IF("LSPGROMBuilderAndSolver", (this->GetEchoLevel() > 2)) << "Number of degrees of freedom:" << BaseType::GetDofSet().size() << std::endl;
+        KRATOS_INFO_IF("LSPGROMBuilderAndSolver", (this->GetEchoLevel() > 2)) << "Finished setting up the dofs" << std::endl;
+
+#ifdef KRATOS_DEBUG
+        // If reactions are to be calculated, we check if all the dofs have reactions defined
+        if (BaseType::GetCalculateReactionsFlag())
+        {
+            for (const auto& r_dof: BaseType::GetDofSet())
+            {
+                KRATOS_ERROR_IF_NOT(r_dof.HasReaction())
+                    << "Reaction variable not set for the following :\n"
+                    << "Node : " << r_dof.Id() << '\n'
+                    << "Dof  : " << r_dof      << '\n'
+                    << "Not possible to calculate reactions." << std::endl;
+            }
+        }
+#endif
+        KRATOS_CATCH("");
+    } 
+    
+    void BuildAndSolve(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart &rModelPart,
+        TSystemMatrixType &A,
+        TSystemVectorType &Dx,
+        TSystemVectorType &b) override
+    {
+        KRATOS_TRY
+        LSPGSystemMatrixType Arom = ZeroMatrix(BaseType::GetEquationSystemSize(), this->GetNumberOfROMModes());
+        LSPGSystemVectorType brom = ZeroVector(BaseType::GetEquationSystemSize());

```

Comment:
These are initialized here and then initialized again inside `BuildROM`. (I notice this is the original ROMBuilderAndSolver has the same issue.

I propose removing this initialization both from here and from ROMBuilderAndSolver::BuildAndSolve. Then we can remove the BuildAndSolve method alltogether from the LSPG builder and solver since the override will no longer be necessary.